### ISSUE TITLE
Loading multiple images? 

### ISSUE THREAD

**Issue by Lukeout** [pixijs/pixi.js#2805]
I am trying to create a game where random objects fall out of the sky in intervals, I have 400 objects to choose from. It's just random. 

But I dont' want to pre-load all of those images and don't want to create 400 variables at the top of my javascript file.... I would rather load them as they are generated in the sky - how can I do this. 

I have a function called loadObject() in here it loads an object. But when I try to use that function in a loop or in setInterval I get an error saying that the variable for the item already exists, so it will only load 1 object on my stage. 

Is there a way to do this so I do not have to declare 400 variables? ... Please let me know I am just a beginner with Pixi. 

I guess basically I am trying to dynamically load things and dynamically add them to the stage. They will fall down to the bottom of the page and then disappear. 

Edit: I see I can do this, 

for (var i = 0; i < 10; i++) {
    stage.addChild(new PIXI.Sprite.fromImage(randomObject()));
}

Where randomObject will return a string that is the path to a random .png file. But I do not have access to those children anymore, correct?

**Comment by whizzkid** [comment]
You might want to look into using a spritesheet. Then you have 1 big image containing all your separate images. 
You load it only once. and reuse as much as you like.

If you load new images while your game is playing, you will probably get jittery gameplay.  It's something you want to avoid.


Your edit is not correct btw. Sprite.fromImage does not need to be NEW'ed.

I've redone your script, so that you can still have access to the images.
Just split it up in separate steps..


```
for (var i = 0; i < 10; i++) {
var newSprite  = PIXI.Sprite.fromImage(randomObject());
stage.addChild(newSprite);
newSprite.position.set(100,200); //  <-- look, you still have access to it :)
}

```

BTW, you can also get access to sprites in any container by knowing where it is in the displaylist.

```
var mySprite = myContainer.getChildAt(0);    // <-- the first sprite in the container.

```

or if you want "the latest sprite(container actually) that was added ": 

`var mySprite = myContainer.getChildAt(myContainer.children.length-1);`

**Comment by englercj** [comment]
Closing this since it seems answered.
